sdk/storage/azure_storage_blob/tests/blob_client.rs

```diff
@@ -644,3 +648,81 @@ async fn test_encoding_edge_cases(ctx: TestContext) -> Result<(), Box<dyn Error>
 
     Ok(())
 }
+
+#[recorded::test]
+async fn test_immutability_policy(ctx: TestContext) -> Result<(), Box<dyn Error>> {
```

Comment:
Testing against a regular Storage account yields:
>HTTP/1.1 400 ImmutableStorageWithVersioning: feature is not enabled.

Testing against versioned _without_ immutable storage on:
><Error><Code>ImmutableStorageWithVersioningIsNotEnabled</Code><Message>ImmutableStorageWithVersioning: feature is not enabled.

So, as a result, had to use a versioned account + immutable storage with versioning enabled.

As a result, having the same issues with deleting the container as `set_legal_hold`:
><Error><Code>ContainerImmutableStorageWithVersioningEnabled</Code><Message>The requested operation is not allowed as the container has a immutable storage with versioning enabled.

Reply:
Mark as playback only for now, I guess. Open an issue to track getting proper clean up on this and the legal hold tests so we can enable them to run Live.

Reply:
Why can't you update your `test-resources.bicep` file? That's the whole reason it's there and you should be using it, just like in all other languages (I wrote those original scripts for **every** Azure SDK language). I'd rather we don't have a confusing concept of "playback-only" tests. That implies you have to record them at some point which means modifying source which means people have to manually interject at least a couple times. Just use a `#[tokio::test]` with a `MockHttpClient` from our `azure_core_test` crate instead. I do that in lots of places. Pretty easy to use.

Reply:
@heaths, we need to mark these tests as playback only because containers with immutability policies (and legal holds) are very difficult to clean up. They require the management plane to do so which we don't have yet in Rust. We will try and figure out a way to delete them and change them to regular recorded tests later. There is nothing that can be changed in the bicep to accomplish this.

We have the concept of playback only in every other language and we use it sparingly for situations like this. It would be nice to have that concept available in Rust. Mocking would require extra effort and means these tests will work and look different than the rest of our tests.